js/calculator.js

```diff
@@ -1,155 +1,269 @@
+                screenValue.value = " "; 
+            })
+            break;
+        case 'equal': //equal
+            if(screenValue.value == " "){
+                screenValue.value = eval(calsi.display.value);
+            }
+            else if(screenValue.value.includes("^")){
+                xPowY();
+            }
+            else{
+                screenValue.value = eval(calsi.display.value);
+            }
+            break;
+        case 'log': //log
+            screenValue.value = Math.log(screenValue.value);
+            break;
+        case 'tenPowerX': //10^x
+            screenValue.value = Math.pow(10,screenValue.value);
+            break;
+        case 'squareRoot': //square root
+            screenValue.value = Math.sqrt(screenValue.value);
+            break;
+        case 'square': //find square
+            screenValue.value = Math.pow(screenValue.value,2);
+            break;
+        case 'oneByX': //find 1/x
+            screenValue.value = 1 / screenValue.value;
+            break;
+        case 'abs': //find |x|
+            screenValue.value = Math.abs(screenValue.value);
+            break;
+        case 'exp': //find exp
+            screenValue.value = Math.exp(screenValue.value);
+            break;
+        case 'FE': //find F-E
+            if(screenValue.value == " ")
+            {
+                screenValue.value = '0.e+0';
+            }
+            else{
+                screenValue.value = screenValue.value + ".e+0";
+            }
+            break;
     }
 }
-//three function
-const three = () =>{
-    if(document.querySelector("#display"). value == " "){
-        document.querySelector("#display").value = "3";
-    }
-    else{
-        document.querySelector("#display").value = document.querySelector("#display").value + "3";
+//factorial find
+function factorial(){
+    let fact = 1;
+    for(let i = 1; i <= screenValue.value; i++){
+      fact *= i;
     }
+    screenValue.value = fact;
 }
-//four function
-const four = () =>{
-    if(document.querySelector("#display"). value == " "){
-        document.querySelector("#display").value = "4";
-    }
-    else{
-        document.querySelector("#display").value = document.querySelector("#display").value + "4";
+//X power Y
+function xPowY(){
+    let a = screenValue.value[screenValue.value.length-1];
+    if(a.match(/[0-9]/)){
+        let b = screenValue.value.split("^");
+        screenValue.value = Math.pow(b[0],b[1]);
     }
 }
-//five function
-const five = () =>{
-    if(document.querySelector("#display"). value == " "){
-        document.querySelector("#display").value = "5";
-    }
-    else{
-        document.querySelector("#display").value = document.querySelector("#display").value + "5";
+//Trigonometry function
+function trigonometry(num){
+    let n;
+    var mathPI = Math.PI /180 ;
+    switch(num){
+        case 'sin':
+            n = Math.sin(screenValue.value *  mathPI)
```

Comment:
Can we do multiplication and store that value in particular variable and then use it everywhere instead of doing multiplication each case statement? 

i.e.
var result =  (screenValue.value *  mathPI) ;

Reply:
issue solve.